### PR TITLE
fix #1385 type alias to tyor type in proc mismatch

### LINKED ISSUE #1385: Distinct generic parameter types
``` nimrod
type intorfloat = int or float # Adding distinct doesn't help

proc x(a: intorfloat; b: intorfloat) =
  echo a, " ", b

# Working
proc y(a: int or float; b: int or float) =
  echo a, " ", b

var c: float = 2.0
var d: int = 3
x(c, d)
```

I would expect this code to work, but get a type mismatch:

```
foo.nim(11, 1) Error: type mismatch: got (float, int)
but expected one of: 
foo.x(a: intorfloat, b: intorfloat)
```

On a related note, maybe this is also supposed to work:

``` nimrod
proc z[T1, T2: int or float](a: T1; b: T2) =
  echo a, " ", b
```

But it seems to suffer from the same problem.

### PR DESCRIPTION
**Pull request by bung87**
fix #1385

### CODE REVIEW

File: compiler/semtypes.nim [hunk 1]
```diff
@@ -1261,6 +1261,14 @@ proc isMagic(sym: PSym): bool =
   let nPragmas = sym.ast[pragmasPos]
   return hasPragma(nPragmas, wMagic)
 
+proc maybeForkType(c: PContext, t: PType): PType =
+  if t.kind in {tyOr} and t.sym != nil:
```
Comment (Araq) [inline compiler/semtypes.nim:1265]: Seems to me you're looking for:

```nim

if t.kind == tyOr: result = copyType(t)
else: result = t

```

Comment (bung87) [reply compiler/semtypes.nim:1265]: this solution does't work. the type used in type equality and symbol binding. one of tyor's son get isNone, so here the symbol binding need extro information to distinguish

Comment (bung87) [reply compiler/semtypes.nim:1265]: use temporary solution mitigate the problem

**Comment by beef331** [comment]
Is this not intended behaviour for named typeclasses? As the issue points to `distinct` works to allow binding to these typeclasses with different types.
```nim
type intorfloat = distinct (int or float)

proc x(a, b: intorfloat) =
  echo a, " ", b

var c: float = 2.0
var d: int = 3
x(c, d)
```

**Comment by bung87** [comment]
> 

oh, distinct works fine, so not a issue now.

**Pull request closed without merging**